### PR TITLE
Implement resource for managing Virtual Clusters Credentials

### PR DESCRIPTION
**Pull request by rprectangle**
- HTTP internal client is extended to support the virtual cluster credentials endpoint.
- Terraform resource `warpstream_virtual_cluster_credentials` is implemented.
- Documentation and usage examples are included.

### CODE REVIEW

**Comment by rprectangle** [comment]
Once merged, I intend to tag `main` with `v0.2.0` to release it on the terraform registry.

File: internal/provider/virtual_cluster_credentials_resource.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+type virtualClusterCredentialsResource struct {
+	client *api.Client
+}
+
+// virtualClusterCredentialsModel maps credentials schema data.
+type virtualClusterCredentialsModel struct {
+	ID               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
+	UserName         types.String `tfsdk:"username"`
+	Password         types.String `tfsdk:"password"`
```
Comment (epot) [inline internal/provider/virtual_cluster_credentials_resource.go:36]: This is the first time I look at the actual implementation of a terraform provider, so maybe it's a stupid question, but does this mean that we are storing the password as part of the tf state?

Comment (rprectangle) [reply internal/provider/virtual_cluster_credentials_resource.go:36]: You are correct, the password is stored in the tf state. If we don't save it, the password would be lost.

At line 110 we declare it as a `sensitive` attribute so terraform will do its best to never display it.
It is not uncommon for terraform state to store sensitive values and the user should use a secured backend in that case.
See https://developer.hashicorp.com/terraform/language/state/sensitive-data.

Comment (epot) [reply internal/provider/virtual_cluster_credentials_resource.go:36]: ok thanks for the pointer 👍

**Review by epot — APPROVED** [review]
LGTM

**Pull request merged**